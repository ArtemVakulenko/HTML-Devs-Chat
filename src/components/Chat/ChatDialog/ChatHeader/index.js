import { connect } from 'react-redux';
import { activeRoom } from 'store/mainPage/selectors';
import { clearActiveRoom, clearMessageValue } from 'store/mainPage/actions';
import ChatHeader from './ChatHeader';

const mapStateToProps = (state) => ({
  activeRoom: activeRoom(state),
});
const mapDispatchToProps = {
  clearActiveRoom, clearMessageValue,
};

export default connect(mapStateToProps, mapDispatchToProps)(ChatHeader);
