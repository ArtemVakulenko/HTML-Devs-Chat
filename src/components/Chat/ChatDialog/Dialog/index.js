import { getMessages } from 'store/mainPage/actions';
import { messages, userId } from 'store/mainPage/selectors';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { compose } from 'redux';
import Dialog from './Dialog';

const mapStateToProps = (state) => ({
  messages: messages(state),
  userId: userId(state),
});
const mapDispatchToProps = (dispatch) => ({
  getMessages: (payload) => dispatch(getMessages(payload)),
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter,
)(Dialog);
