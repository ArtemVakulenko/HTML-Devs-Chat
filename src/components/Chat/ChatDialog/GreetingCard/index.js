import { connect } from 'react-redux';
import { userName } from 'store/mainPage/selectors';
import GreetingCard from './GreetingCard';

const mapStateToProps = (state) => ({
  userName: userName(state),
});

export default connect(mapStateToProps, null)(GreetingCard);
