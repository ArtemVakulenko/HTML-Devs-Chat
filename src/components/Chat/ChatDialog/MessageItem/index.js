import { userName, getUserPic } from 'store/mainPage/selectors';
import { connect } from 'react-redux';
import { getOneUser } from 'store/modals/actions';
import MessageItem from './MessageItem';

const mapStateToProps = (state) => ({
  userName: userName(state),
  userPic: getUserPic(state),
});

const mapDispatchToProps = {
  getOneUser,
};
export default connect(mapStateToProps, mapDispatchToProps)(MessageItem);
