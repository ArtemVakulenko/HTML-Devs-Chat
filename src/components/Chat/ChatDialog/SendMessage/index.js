import { connect } from 'react-redux';
import { messageValue, userName, getUserPic } from 'store/mainPage/selectors';
import { setMessageValue, sendMessage } from 'store/mainPage/actions';
import { withRouter } from 'react-router';
import { compose } from 'redux';
import SendMessage from './SendMessage';

const mapStateToProps = (state) => ({
  messageValue: messageValue(state),
  userName: userName(state),
  userPic: getUserPic(state),
});
const mapDispatchToProps = (dispatch) => ({
  setMessageValue: (payload) => dispatch(setMessageValue(payload)),
  sendMessage: (payload) => dispatch(sendMessage(payload)),
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter,
)(SendMessage);
