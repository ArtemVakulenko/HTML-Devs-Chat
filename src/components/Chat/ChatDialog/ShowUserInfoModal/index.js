import { connect } from 'react-redux';
import { getUserToShow } from 'store/modals/selectors';
import ShowUserInfoModal from './ShowUserInfoModal';

const mapStateToProps = (state) => ({
  userToShow: getUserToShow(state),
});

export default connect(mapStateToProps, null)(ShowUserInfoModal);
