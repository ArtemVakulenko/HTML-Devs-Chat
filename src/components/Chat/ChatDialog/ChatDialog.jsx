import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import { StChatDialogContainer } from './styled';
import ChatHeader from './ChatHeader';
import SendMessage from './SendMessage';
import Dialog from './Dialog';
import GreetingCard from './GreetingCard';

const ChatDialog = ({ match }) => {
  const { colors, theme } = useTheme();
  const path = match.url;
  return path === '/main'
  ? <GreetingCard/>
  : (
    <StChatDialogContainer colors={colors} theme={theme}>
      <ChatHeader/>
        <Dialog/>
      <SendMessage/>
    </StChatDialogContainer>
  );
};

ChatDialog.propTypes = {
  match: PropTypes.object,
};

export default ChatDialog;
