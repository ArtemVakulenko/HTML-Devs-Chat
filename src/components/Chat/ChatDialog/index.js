import { withRouter } from 'react-router';
import ChatDialog from './ChatDialog';

export default withRouter(ChatDialog);
