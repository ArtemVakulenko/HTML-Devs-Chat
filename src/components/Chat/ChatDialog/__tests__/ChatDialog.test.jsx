import React from 'react';
import configureStore from 'redux-mock-store';
import ChatDialog from '../ChatDialog';
import { shallowSmart } from '../../../../__tests__/testHelper';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({ mainPage: {
  sidebar: {
    search: '',
    rooms: [],
  },
  chat: {
    messages: [],
    roomId: '',
    messageValue: '',
    activeRoom: {},
  },
  user: {
    id: '',
    login: '',
    name: '',
    lastName: '',
    img: '',
    hobbie: '',
    city: '',
    age: '',
  },
} });

jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('ChatDialog', () => {
  describe('123', () => {
    const props = {
      match: {},
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<ChatDialog {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
  });
});
