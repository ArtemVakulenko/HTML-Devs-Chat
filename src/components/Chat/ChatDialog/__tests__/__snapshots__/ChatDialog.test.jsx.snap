// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatDialog 123 Should match snapshot 1`] = `"<div class=\\"styled__StChatDialogContainer-hAlvwF ctssmL\\"><div class=\\"styled__StChatDialogHeader-frWLNJ iLTYqh\\"><div class=\\"styled__StChatTitle-erdfFm fRrVrb\\"></div><button class=\\"styled__StIcon-chPeWD giwZPn\\"></button></div><div class=\\"styled__StDialogContainer-hphmKj kldXao\\"><div></div></div><div class=\\"styled__StSendMessageContainer-lnwVcU gYXjlf\\"><textarea class=\\"styled__StTextArea-cYwHWS hmAZYP\\"></textarea><button class=\\"styled__StIcon-chPeWD dNDmRl\\"></button></div></div>"`;

exports[`LoginPage 123 Should match snapshot 1`] = `"<div class=\\"styled__StChatDialogContainer-hAlvwF ctssmL\\"><div class=\\"styled__StChatDialogHeader-frWLNJ iLTYqh\\"><div class=\\"styled__StChatTitle-erdfFm fRrVrb\\"></div><button class=\\"styled__StIcon-chPeWD giwZPn\\"></button></div><div class=\\"styled__StDialogContainer-hphmKj kldXao\\"><div></div></div><div class=\\"styled__StSendMessageContainer-lnwVcU gYXjlf\\"><textarea class=\\"styled__StTextArea-cYwHWS hmAZYP\\"></textarea><button class=\\"styled__StIcon-chPeWD dNDmRl\\"></button></div></div>"`;
